gax-java/gax-grpc/src/main/java/com/google/api/gax/grpc/InstantiatingGrpcChannelProvider.java

```diff
@@ -225,6 +254,40 @@ public TransportChannelProvider withEndpoint(String endpoint) {
     return toBuilder().setEndpoint(endpoint).build();
   }
 
+  /**
+   * Specify the MTLS endpoint.
+   *
+   * <p>The value of {@code mtlsEndpoint} must be of the form {@code host:port}.
+   *
+   * @param mtlsEndpoint
+   * @return A new {@link InstantiatingGrpcChannelProvider} with the specified MTLS endpoint
+   *     configured
+   */
+  @Override
+  public TransportChannelProvider withMtlsEndpoint(String mtlsEndpoint) {
+    if (!mtlsEndpoint.isEmpty()) {
+      validateEndpoint(mtlsEndpoint);
+    }
+    return toBuilder().setMtlsEndpoint(mtlsEndpoint).build();
+  }
```

Comment:
I believe we usually allow configuration via clientsettings/ stubsettings: https://github.com/googleapis/sdk-platform-java/blob/c8e3941ef6f5bd1236f5ceedfd488e5113928471/gax-java/gax/src/main/java/com/google/api/gax/rpc/ClientSettings.java#L270-L273

I would prefer to not add it in the TransportChannelProvider if possible.

Reply:
Ok, I see from below that the mtlsEndpoint is passed to the TransportChannelProvider. I think that makes sense given that we do that for endpoint right now. 

I am also thinking that we should also add a similar setter to the settings classes linked above for consistency.

Reply:
Done, added mtlsEndpoint setter/getter to the ClientSettings in https://github.com/googleapis/sdk-platform-java/pull/3326/commits/28adb31ba9fe6ff5d5249189c776acc39c6ca503

Reply:
I know this wasn't the case for other getters and setters, but can you add some simple javadocs to those public methods? 

The other public methods are legacy code and we haven't gotten around to fixing them.

Reply:
Added javadocs to the public methods in ClientSettings in https://github.com/googleapis/sdk-platform-java/pull/3326/commits/035e17eb56a4c4d74b31994df889593b87e61192